rasa/core/agent.py

```diff
@@ -888,6 +888,7 @@ def create_processor(
             self.policy_ensemble,
             self.domain,
             self.tracker_store,
+            self.lock_store,
```

Comment:
not a fan of handing down intrastructure components to the `processor` but should in my opinion addressed separately (e.g. as part of https://github.com/RasaHQ/rasa/issues/5257)

Reply:
I couldn't think of a better way to do this without requiring re-factoring. Is there a better approach to this than passing it into processor?

Reply:
Also can't think of one for now. I think it's fine for now 👍🏻